### PR TITLE
Feature/implementation identity

### PR DESCRIPTION
**Pull request by Dinek007**

### CODE REVIEW

File: src/renderer/store/handlers/users.ts [hunk 1]
```diff
@@ -151,6 +153,9 @@ export const createOrUpdateUser = (payload: {
   const address = identitySelector.address(getState())
   const privKey = identitySelector.signerPrivKey(getState())
 
+  await dispatch(certificatesHandlers.epics.createOwnCertificate(nickname))
```
Comment (kowalski) [inline src/renderer/store/handlers/users.ts:156]: after you refactor this to saga, the code here should be just:
```
dispatch(certificatesActions.creactOwnCertificate(nickname))
```

so in certificates master saga you should have: 
```
takeEvery(certificatesActions.creactOwnCertificate.type, creactOwnCertificate)
```
`creactOwnCertificate` here is a saga that call allt the crypto functions and finnaly does:

```
yield *put(certificatesActions.saveCertificate(ownCertificate))
```

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/renderer/store/certificates/certificates.saga.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import { certificatesActions, certificatesReducer, CertificatesState } from './certificates.reducer'
+import { creactOwnCertificate } from './certificates.saga'
+import { createUserCsr } from '../../pkijs/generatePems/requestCertificate'
+import { createUserCert } from '../../pkijs/generatePems/generateUserCertificate'
+
+describe('checkCertificatesSaga', () => {
+  test('creating own cert', () => {
+    void expectSaga(creactOwnCertificate, { payload: 'name', type: certificatesActions.creactOwnCertificate.type })
+      .withReducer((certificatesReducer), {
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.test.ts:11]: Shiyld be:

combineReducers({ [StoreKeys.certificates]: certificatesReducers })

This way you have the same structure in tests as in app, so your selectors work normally

File: src/renderer/store/certificates/certificates.saga.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export function* creactOwnCertificate(
+  action: PayloadAction<ReturnType<typeof certificatesActions.creactOwnCertificate>['payload']>
+): Generator {
+  const certString = dataFromRootPems.certificate
+  const keyString = dataFromRootPems.privKey
+
+  const notBeforeDate = yield call(getDate)
+  const notAfterDate = yield call(getDate, '1/1/2031')
+
+  yield apply(electronStore, electronStore.get, ['hiddenServices'])
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:29]: Thos is to get onion address few lines below.

You're still calliny there electroStore.get without yieldong effect

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/renderer/store/certificates/certificates.saga.ts [hunk 4]
```diff
@@ -15,28 +15,39 @@ export function* responseGetCertificates(
...
-    peerId: electronStore.get('peerId')
+    commonName: hiddenServices.libp2pHiddenService.onionAddress,
+    peerId: yield apply(electronStore, electronStore.get, ['peerId'])
   }
 
   const user = yield* call(createUserCsr, userData)
-  const userCertData = yield* call(createUserCert, certString, keyString, user.userCsr, notBeforeDate, notAfterDate)
 
+  const userCertData = yield* call(createUserCert, certString, keyString, user.userCsr, notBeforeDate, notAfterDate)
+  console.log({ userCertData })
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:50]: console.log

File: src/renderer/store/certificates/certificates.saga.ts [hunk 5]
```diff
@@ -15,28 +15,39 @@ export function* responseGetCertificates(
...
+    libp2pHiddenService?: {
+      onionAddress: string
+      privateKey: string
+    }
+  }
+
+  const hiddenServices: HiddenServicesType = yield apply(
+    electronStore,
+    electronStore.get,
+    ['hiddenServices']
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:38]: I think it should be 
```

 const hiddenServices: HiddenServicesType = yield apply(    electronStore,    electronStore.get,    'hiddenServices'  )
 
 ```
 
 (without the wrapping in list)

File: src/renderer/store/certificates/certificates.saga.ts [hunk 6]
```diff
@@ -15,28 +15,39 @@ export function* responseGetCertificates(
...
 
-  yield apply(electronStore, electronStore.get, ['hiddenServices'])
+  interface HiddenServicesType {
+    libp2pHiddenService?: {
+      onionAddress: string
+      privateKey: string
+    }
+  }
+
+  const hiddenServices: HiddenServicesType = yield apply(
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:35]: import `apply` from `typed-redux-saga` and change `yield` to `yield*`

File: src/renderer/store/certificates/certificates.saga.ts [hunk 7]
```diff
@@ -15,28 +15,39 @@ export function* responseGetCertificates(
...
+    electronStore.get,
+    ['hiddenServices']
+  )
 
   const userData = {
     zbayNickname: action.payload,
-    commonName: electronStore.get('hiddenServices').libp2pHiddenService.onionAddress,
-    peerId: electronStore.get('peerId')
+    commonName: hiddenServices.libp2pHiddenService.onionAddress,
+    peerId: yield apply(electronStore, electronStore.get, ['peerId'])
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:44]: 'peerId' not [`'peerId']

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/renderer/store/certificates/certificates.saga.ts (hunk 5 above)
Comment (Dinek007) [reply src/renderer/store/certificates/certificates.saga.ts:38]: image

Comment (Dinek007) [reply src/renderer/store/certificates/certificates.saga.ts:38]: i found it in the documentation

Comment (kowalski) [reply src/renderer/store/certificates/certificates.saga.ts:38]: Ok, ok than.

**Review by kowalski — APPROVED** [review]

**Pull request merged**